### PR TITLE
chore(Downshift): update dependency to latest

### PR DESCRIPTION
**Pull request by dakahn**
Updates our Downshift dependency to latest (v6.1.7)

_from downshift_

BREAKING CHANGES

    Update TS typings for selectedItem to accept null in both useSelect and useCombobox.

To migrate to the new change, update your types or code if necessary. selectedItem, defaultSelectedItem and initialSelectedItem now have Item | null instead of Item type. PR with the changes: https://github.com/downshift-js/downshift/pull/1090

    Update TS typings for itemToString to accept null for the item parameter, in useSelect and useCombobox + in Downshift where this was missing. useMultipleSelection type for itemToString stays the same as it can't receive null as item.

To migrate to the new change, update your types or code if necessary. itemToString: (item: Item) => string -> itemToString: (item: Item | null) => string}. PR with the changes: https://github.com/downshift-js/downshift/pull/1075 https://github.com/downshift-js/downshift/pull/1105

    Pass type to the onChange (onInputValueChange, onHighlightedIndexChange, onSelectedItemChange, onIsOpenChange) handler parameters, as specified in the documentation. Also updated the TS typings to reflect this + onStateChange - the type parameter was passed but it was not reflected in the TS types.

To migrate to the new change, update your types or code if necessary, better to view the changes in the PR: https://github.com/downshift-js/downshift/pull/985. Important: please update to the 6.0.2 version since it contains a couple of fixes for the changes in this Breaking Change. Final changes:

  stateReducer?: (
    state: UseComboboxState<Item>,
    actionAndChanges: UseComboboxStateChangeOptions<Item>, // UseComboboxStateChangeOptions has the correct typings
  ) => Partial<UseComboboxState<Item>> // this now reflects the correct return, which is the partial state.
  onSelectedItemChange?: (changes: UseComboboxStateChange<Item>) => void // changes have partial state + type
  onIsOpenChange?: (changes: UseComboboxStateChange<Item>) => void 
  onHighlightedIndexChange?: (changes: UseComboboxStateChange<Item>) => void
  onStateChange?: (changes: UseComboboxStateChange<Item>) => void
  onInputValueChange?: (changes: UseComboboxStateChange<Item>) => void

where

export interface UseComboboxStateChangeOptions<Item>
  extends UseComboboxDispatchAction<Item> { // type and optional parameters needed to compute the next state
  changes: Partial<UseComboboxState<Item>> // partial state changes proposed by the hook
}

export interface UseComboboxDispatchAction<Item> {
  type: UseComboboxStateChangeTypes
  shiftKey?: boolean
  getItemNodeFromIndex?: (index: number) => HTMLElement
  inputValue?: string
  index?: number
  highlightedIndex?: number
  selectedItem?: Item | null
  selectItem?: boolean
}

export interface UseComboboxStateChange<Item>
  extends Partial<UseComboboxState<Item>> {
  type: UseComboboxStateChangeTypes
}

BREAKING BEHAVIOURS

    [useCombobox]: When an item is highlighted by keyboard and user closes the menu using mouse/touch, the item is not selected anymore. The only selection on Blur happens using either Tab / Shift+Tab. PR with the changes: https://github.com/downshift-js/downshift/pull/1109

    [useCombobox & downshift]: When pressing Escape and the menu is open, only close the menu. When the menu is closed and there is an item selected and/or text in the input, clear the selectedItem and the inputValue. PR with the changes: https://github.com/downshift-js/downshift/issues/719

closes #11331

### CODE REVIEW

File: packages/react/package.json [hunk 1]
```diff
@@ -50,7 +50,7 @@
     "@carbon/telemetry": "0.1.0",
     "classnames": "2.3.1",
     "copy-to-clipboard": "^3.3.1",
-    "downshift": "5.2.1",
+    "downshift": "6.1.7",
```
Comment (joshblack) [inline packages/react/package.json]: Are there any user-facing breaking changes from doing a major update of `downshift`?

Comment (dakahn) [reply packages/react/package.json]: updated the PR description with their changelog 👀

**Comment by jnm2377** [comment]
How should we review this?

**Comment by dakahn** [comment]
@jnm2377 sorry, for the lack of clarity there. Basically, we're updating a dependency for Dropdown, Combobox and Multiselect. So running through basically functionality, testing, and keyboard/screen reader functionality is preferable

**Comment by dakahn** [comment]
Looks like updating breaks tooltips for truncated menu items in Chrome/Chromium browsers like [we're currently seeing in Firefox](https://github.com/carbon-design-system/carbon/issues/8864), which is interesting.

Could be related to the ref weirdness we're seeing in our `useIsTruncated` hook

**Review by dakahn — COMMENTED** [review]
Looks like updating breaks tooltips for truncated menu items in Chrome/Chromium browsers like [we're currently seeing in Firefox](https://github.com/carbon-design-system/carbon/issues/8864), which is interesting.

Could be related to the ref weirdness we're seeing in our `useIsTruncated` hook

**Comment by jnm2377** [comment]
> Looks like updating breaks tooltips for truncated menu items in Chrome/Chromium browsers like [we're currently seeing in Firefox](https://github.com/carbon-design-system/carbon/issues/8864), which is interesting.
> 
> Could be related to the ref weirdness we're seeing in our `useIsTruncated` hook

@dakahn is this blocked until we figure that out, then? not being able to read truncated text fully seems like an important issue, even though I think essential functionality (selecting items) still works. Idk. thoughts?

**Comment by jnm2377** [comment]
@dakahn just wanted to follow up on this. Do we think we should defer updating since it breaks truncated tooltips?

**Comment by dakahn** [comment]
yeah, until we get a handle on the existing bug we're aware of this is officially blocked id say

**Comment by tw15egan** [comment]
Did a quick run-through of all the downshift components, and the only issue I saw was that I was able to keyboard navigate to and select a disabled item in `Multiselect`. All other elements seemed like they behaved correctly. Also noticed an issue with the `With Initial Selected Items` story, with a `disabled` initially selected. Not sure if that is intended. 

Tabbing / selecting disabled item
Screen Shot 2022-08-25 at 1 34 57 PM

Initially selected item is disabled
Screen Shot 2022-08-25 at 1 35 38 PM

**Pull request closed without merging**